raiden/api/python.py

```diff
@@ -1696,13 +1696,39 @@ def create_light_client_payment(
         amount: typing.TokenAmount,
         secrethash: typing.SecretHash
     ) -> HubResponseMessage:
+
         channel_state = views.get_channelstate_for(
             views.state_from_raiden(self.raiden),
             registry_address,
             token_address,
             creator_address,
             partner_address,
         )
+        #If we dont have a channel with the partner we need to get a channel to try a mediated transfer
+        if not channel_state:
+            token_network_id = views.get_token_network_by_token_address(
+                views.state_from_raiden(self.raiden), registry_address, token_address
+            )
+            routes, _ = routing.get_best_routes(
+                chain_state=views.state_from_raiden(self.raiden),
+                token_network_id=token_network_id.address,
+                one_to_n_address=self.raiden.default_one_to_n_address,
+                from_address=InitiatorAddress(creator_address),
+                to_address=partner_address,
+                amount=amount,
+                previous_address=None,
+                config=self.raiden.config,
+                privkey=self.raiden.privkey,
+            )
+            if len(routes) > 0:
+                channel_state = views.get_channelstate_for(
+                    views.state_from_raiden(self.raiden),
+                    registry_address,
+                    token_address,
+                    creator_address,
+                    routes[0].node_address,
```

Comment:
Here we are returning the first route to the light client, to make it signing the transaction with that `channel_identifier`. I have a question about that: what happens if the routes are different when sending the signed locked transfer? 

When the LC sends the signed LT, the `ActionInitInitiatorLight ` state change is processed, and that state change receives routes, that are being calculated again at that point.

Basically, `initiate_payment ` triggers again the route calculation